exporter/elasticsearchexporter/README.md

```diff
@@ -139,8 +139,9 @@ where `data_stream.type` is `logs` for log records, `metrics` for data points, a
 In a special case with `mapping::mode: bodymap`, `data_stream.type` field (valid values: `logs`, `metrics`) can be dynamically set from attributes.
 The resulting documents will contain the corresponding `data_stream.*` fields, see restrictions applied to [Data Stream Fields](https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html).
    1. `data_stream.dataset` or `data_stream.namespace` in attributes (precedence: log record / data point / span attribute > scope attribute > resource attribute)
-   2. Otherwise, if scope name matches regex `/receiver/(\w*receiver)`, `data_stream.dataset` will be capture group #1
-   3. Otherwise, `data_stream.dataset` falls back to `generic` and `data_stream.namespace` falls back to `default`. 
+   2. Otherwise, if a scope attribute with the name `encoding.format` exists and contains a string value, `data_stream.dataset` will be set to this value. 
```

Comment:
Isn't the addition of this quite arbitrary? What if users want to use another scope/resource attribute for routing? Also what if there are components that already populate this attribute? Hardcoding this here will break the routing for them, right? In this, why not to make this configurable so as users can define the routing rule based on whatever they would like to i.e. `routing_attribute: resource.attributes["someattr"]` (similarly to ottl)

Reply:
can't they already effectively do this just by using the explicit `data_stream.*` attributes? 

> Also what if there are components that already populate this attribute?

this is a fair point, but we're not currently aware of any. 

> why not to make this configurable so as users can define the routing rule based on whatever they would like 

the intention here is to provide a default behaviour so encoding extension + es exporter works OOTB.

Reply:
Note that any changes to the routing logic also need to be applied to the Elasticsearch OTLP endpoint. 
For this proposed change, I’m a bit worried that the attribute is too specific about encoding extensions. It feels like a slippery slope for adding more similar routing attributes. Can we instead either set an appropriate scope name in the encoding extensions or introduce a more generic concept, something like a dataset or schema name that describes the shape of the data that we can use for routing?